msg_id: eliminate the use of bind in msg_id parser

seliopou opened the pull request.

seliopou committed:
msg_id: eliminate the use of bind in msg_id parser

dinosaure commented:
You miss the `@` in the middle and the `>` at the end. Then `msg_id` was come from the RFC822 but it's not related to the `Content-Type`'s value.

seliopou committed:
msg_id: add overlooked `char '@'` parser

Also reorganize to make the whitespace parsing clearer, and separate
from the parsing of the actual value.

seliopou commented:
Sorry, fixed!

seliopou committed:
msg_id: add overlooked trailing '>'
